src/HydraAuction/Tx/Common.hs

```diff
@@ -292,6 +299,32 @@ scriptUtxos script terms = do
   scriptAddress' <- scriptAddress script terms
   liftIO $ queryUTxO networkId nodeSocket QueryTip [scriptAddress']
 
+getStadingBidDatum :: UTxO.UTxO -> StandingBidDatum
+getStadingBidDatum standingBidUtxo =
+  case UTxO.pairs standingBidUtxo of
+    [(_, out)] -> case txOutDatum out of
+      TxOutDatumInline scriptData ->
+        case fromData $ toPlutusData scriptData of
+          Just standingBidDatum -> standingBidDatum
+          Nothing ->
+            error "Impossible happened: Cannot decode standing bid datum"
+      _ -> error "Impossible happened: No inline data for standing bid"
+    _ -> error "Wrong number of standing bid UTxOs found"
+
+currentWinningBidder :: AuctionTerms -> Runner (Maybe PubKeyHash)
```

Comment:
It is strange that this function was in `Escrow`, but still is not `StandingBid.hs` better place for it? Cuz it is tied to logic of StagingBid. Same with other two.

Reply:
Unfortunately that gives rise to a circular dependency :(

Reply:
You are meaning Commons -> StandingBid dep? Why? I am talking about moving all three functions there and there is no other deps from Commons as far as I see.

Reply:
Because `Escrow` uses those functions, but `StandingBid` imports `Escrow`